engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/ingress/IngressPreparer.scala

```diff
@@ -0,0 +1,60 @@
+import pl.touk.nussknacker.k8s.manager.ingress.IngressPreparer.rewriteAnnotation
+import play.api.libs.json.Json
+import skuber.networking.v1.Ingress
+import monocle.std.option.some
+
+case class IngressConfig(host: String, config: Config = ConfigFactory.empty())
+
+class IngressPreparer(config: IngressConfig, nuInstanceName: Option[String]) {
+
+  def prepare(processVersion: ProcessVersion, typeSpecificData: TypeSpecificData, serviceName: String, servicePort: Int): Option[Ingress] =
+    typeSpecificData match {
+      case _: LiteStreamMetaData => None
+      case rrMetaData: RequestResponseMetaData => Some(prepareRequestResponseIngress(processVersion, rrMetaData, serviceName, servicePort))
+      case other => throw new IllegalArgumentException("Not supported scenario meta data type: " + other)
+    }
+
+  private def prepareRequestResponseIngress(processVersion: ProcessVersion, rrMetaData: RequestResponseMetaData, serviceName: String, servicePort: Int): Ingress = {
+    val objectName = IngressPreparer.name(nuInstanceName, s"scenario-${processVersion.processName.value}")
+    val labels = labelsForScenario(processVersion, nuInstanceName)
+    val slug = RequestResponseSlugUtils.determineSlug(processVersion.processName, rrMetaData, nuInstanceName)
+
+    //we use 'OptionOptics some' here and do not worry about withDefault because _.spec is provided in defaultMinimalIngress
+    val ingressSpecLens = GenLens[Ingress](_.spec) composePrism some
+
+    val lens = GenLens[Ingress](_.metadata.name).set(objectName) andThen
+      GenLens[Ingress](_.metadata.labels).modify(_ ++ labels) andThen
+      GenLens[Ingress](_.metadata.annotations).modify(_ ++ rewriteAnnotation) andThen
+      (ingressSpecLens composeLens GenLens[Ingress.Spec](_.rules)).modify(_ ++ List(
+        Ingress.Rule(Some(config.host), Ingress.HttpRule(paths = List(Ingress.Path(
+          path = s"/$slug" + "(/|$)(.*)",
+          backend = Ingress.Backend(Some(Ingress.ServiceType(serviceName, Ingress.Port(number = Some(servicePort))))),
+          pathType = Ingress.PathType.Prefix))))))
+
+    lens(fromUserConfig)
+  }
+
+  private def fromUserConfig: Ingress = {
+    val minimalConfig = ConfigFactory.parseURL(getClass.getResource("/defaultMinimalIngress.conf"))
+    val finalConfig = config.config.withFallback(minimalConfig)
+    Json.parse(finalConfig.root().render(ConfigRenderOptions.concise())).as[Ingress]
+  }
+}
+
+object IngressPreparer {
+  private[ingress] def name(nussknackerInstanceName: Option[String], scenarioName: String): String =
+    sanitizeObjectName(nameWithoutSanitization(nussknackerInstanceName, scenarioName))
+
+  private[ingress] def nameWithoutSanitization(nussknackerInstanceName: Option[String], scenarioName: String): String =
+    objectNamePrefixedWithNussknackerInstanceNameWithoutSanitization(nussknackerInstanceName, scenarioName)
+
+  private[ingress] val rewriteAnnotation = Map("nginx.ingress.kubernetes.io/rewrite-target" -> "/$2")
```

Comment:
I think that it is not necessary - we can just write in docs that currently we assume that ingress controller is nginx, we use some nginx specific annotations and because of that something may not work correctly for other controllers. WDYT? Because controller type doesn't change anything, still someone need  to take a look into documentation to find this controller type setting. TL;DR YAGNI

Reply:
When the `RR on K8s` docs are ready, I put a not there, that we only support Nginx Ingress Controller for now.